retrieve-image-tags/retrieve-image-tags.py

```diff
@@ -109,63 +192,92 @@ def _get_image_tags(registry, namespace, package, nexturl, page=1):
 filetext = textfile.read()
 textfile.close()
 
-# Function to search textfile content
-def image_tag_already_exist(filetext, image, tag):
-    matches = re.findall(fr"{image}\s.*?\s{tag}\n", filetext)
-    if matches:
-        return True
-    return False
-
 alldict = {}
 
 # loop through each repository
 for (key, values) in data.items():
     alldict[key] = {}
     # determine version source for the image tags
-    repoString = values['versionSource'].split(":")
-    if values['versionSource'] != "registry":
+    repoString = values['versionSource'].split(":", 1)
+    if values['versionSource'] != 'registry' and values['versionSource'] != 'helm-oci':
         if len(repoString) != 2:
             print(f"Can not extract proper version source from versionSource {values['versionSource']} for image {key}")
             sys.exit(1)
 
     found_releases = []
+    image_denylist = []
 
     match repoString[0]:
-        case "github-releases":
+        case 'helm-latest' | 'helm-oci':
+            # check if helm binary is available
+            helm_path = shutil.which('helm')
+
+            if helm_path is None:
+                print("no executable found for command 'helm'")
+                sys.exit(1)
+
+            if repoString[0] == 'helm-latest':
+                helm_repo = repoString[1]
+            repo_name = key
+            if helm_repo.startswith('https://'):
+                subprocess.check_call(f"helm repo add {repo_name} {helm_repo}",
+                    shell=True,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.STDOUT)
+                subprocess.check_call('helm repo update',
+                    shell=True,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.STDOUT)
+            if 'imageDenylist' in values:
+                image_denylist = values['imageDenylist']
+            for chart in values['helmCharts']:
+                if 'helmArgs' in values['helmCharts'][chart]:
+                    for arg_key in values['helmCharts'][chart]['helmArgs']:
+                        helm_config_args = values['helmCharts'][chart]['helmArgs'][arg_key]
+                        helm_args = ["--set " + value for value in helm_config_args]
```

Comment:
The name would suggest that you can pass through any helm arg, but this appears to be limited to just setting chart values. Should it be called `helmChartValues` instead?

Reply:
Agree, I changed the structure and naming for this and the `--kube-version` from below.